packages/node-core/src/indexer/connectionPool.service.ts

```diff
@@ -62,12 +62,25 @@ export class ConnectionPoolService<T extends IApiConnectionSpecific<any, any, an
     }
   }
 
+  async updateConnection(api: T, index: number): Promise<void> {
+    if (index >= this.allApi.length) {
+      throw new Error(`Attempting to update connection that does not exist.`);
+    }
+
+    this.allApi[index] = api;
```

Comment:
Do you think this is safe? What happens if the wrong index is provided?

Reply:
1.) the order of the api will be different for different workers, making the connection pool state invalid because only the index is used to reference api. 
2.) one api might overwrite another

so, it's important that we get the index right.

Reply:
Yes its important to get the index right. Is using an int the right thing to key these by? If we use the endpoint it would be much safer